src/EFCore/Metadata/IConventionEntityType.cs

```diff
@@ -42,20 +42,36 @@ public interface IConventionEntityType : IReadOnlyEntityType, IConventionTypeBas
     /// </summary>
     bool IsKeyless { get; }
 
+    /// <summary>
+    ///     Sets the query filter automatically applied to queries for this entity type.
+    /// </summary>
+    /// <param name="queryFilter">The query filter.</param>
+    /// <returns>The configured filter.</returns>
+    /// <param name="fromDataAnnotation">Indicates whether the configuration was specified using a data annotation.</param>
+    /// <returns>The configured filter.</returns>
+    IQueryFilter? SetQueryFilter(IQueryFilter queryFilter, bool fromDataAnnotation = false);
```

Comment:
Here and in the other related API, keep the previous signature, we don't want the implementation details to leek out like this. Let the implementation encapsulate how the configuration is stored.

```suggestion
    IQueryFilter? SetQueryFilter(string filterKey, LambdaExpression? queryFilter, bool fromDataAnnotation = false);
```

Reply:
this will lead us back to the convention validation issues. Either we will add it to the exception in APIConsistencyTest or the IQueryFilter will need to be added to MetadataTypes which means that it will have to implement all sorts of other interfaces like IConventionAnnotatable, IMutableAnnotatable and IReadOnlyAnnotatable, which I think will be an overkill for this effort here